contracts/services/template/src/msg.rs

```diff
@@ -1,56 +1,52 @@
 use cosmwasm_schema::{cw_serde, QueryResponses};
 use cosmwasm_std::{Addr, Deps, DepsMut};
 use cw_ownable::cw_ownable_query;
-use valence_macros::OptionalStruct;
-use valence_service_utils::{
-    error::ServiceError, msg::ServiceConfigValidation, ServiceConfigInterface,
-};
+use valence_macros::{valence_service_query, ValenceServiceInterface};
+use valence_service_utils::{error::ServiceError, msg::ServiceConfigValidation};
 
 #[cw_serde]
 pub enum ActionsMsgs {
     NoOp {},
 }
 
+#[valence_service_query]
 #[cw_ownable_query]
 #[cw_serde]
 #[derive(QueryResponses)]
 /// Enum representing the different query messages that can be sent.
-pub enum QueryMsg {
-    /// Query to get the processor address.
-    #[returns(Addr)]
-    GetProcessor {},
-    /// Query to get the service configuration.
-    #[returns(Config)]
-    GetServiceConfig {},
-}
+pub enum QueryMsg {}
 
 #[cw_serde]
-#[derive(OptionalStruct)]
-pub struct ServiceConfig {}
+#[derive(ValenceServiceInterface)]
+pub struct ServiceConfig {
+    /// We ignore this field when generating the ValenceServiceInterface
+    /// This means this field is not updatable
+    #[skip_update]
+    pub skip_update_admin: String,
+    pub optional: Option<String>,
+    pub optional2: String,
+}
 
 impl ServiceConfigValidation<Config> for ServiceConfig {
     #[cfg(not(target_arch = "wasm32"))]
     fn pre_validate(&self, _api: &dyn cosmwasm_std::Api) -> Result<(), ServiceError> {
         Ok(())
     }
 
-    fn validate(&self, _deps: Deps) -> Result<Config, ServiceError> {
-        Ok(Config {})
-    }
-}
-
-impl ServiceConfigInterface<ServiceConfig> for ServiceConfig {
-    /// This function is used to see if 2 configs are different
-    fn is_diff(&self, other: &ServiceConfig) -> bool {
-        !self.eq(other)
+    fn validate(&self, deps: Deps) -> Result<Config, ServiceError> {
+        Ok(Config {
+            admin: deps.api.addr_validate(&self.skip_update_admin)?,
+        })
     }
 }
 
-impl OptionalServiceConfig {
+impl ServiceConfigUpdate {
     pub fn update_config(self, _deps: &DepsMut, _config: &mut Config) -> Result<(), ServiceError> {
         Ok(())
     }
 }
 
 #[cw_serde]
-pub struct Config {}
+pub struct Config {
+    pub admin: Addr,
+}
```

Comment:
should probably added the optional ones here too right? For it to make perfect sense to add other fields to update.